modules/apps/users-admin/users-admin-impl/src/main/java/com/liferay/users/admin/internal/exportimport/data/handler/OrganizationStagedModelDataHandler.java

```diff
@@ -106,16 +106,28 @@ public String getDisplayName(Organization organization) {
 	public boolean validateReference(
 		PortletDataContext portletDataContext, Element referenceElement) {
 
+		Map<Long, Long> organizationIds =
+			(Map<Long, Long>)portletDataContext.getNewPrimaryKeysMap(
+				Organization.class);
+
+		long organizationId = GetterUtil.getLong(
+			referenceElement.attributeValue("class-pk"));
+
+		if (organizationIds.containsKey(organizationId)) {
+			return true;
+		}
+
 		Organization organization =
 			_organizationLocalService.fetchOrganizationByUuidAndCompanyId(
 				GetterUtil.getString(referenceElement.attributeValue("uuid")),
-				GetterUtil.getLong(
-					referenceElement.attributeValue("company-id")));
+				GetterUtil.getLong(portletDataContext.getCompanyId()));
 
 		if (organization == null) {
 			return false;
 		}
 
+		organizationIds.put(organizationId, organization.getOrganizationId());
+
```

Comment:
Is this put necessary? `organizationIds` is never used anywhere else no?

Reply:
While not strictly necessary, we do end up retrieving the information in BlogsAggregatorExportImportPortletPreferencesProcessor when we update our import portlet preferences and therefore this theoretically saves us a step there.

Reply:
Okay that works.